src/chainstate/coordinator/mod.rs

```diff
@@ -845,12 +1196,156 @@ impl<
                         );
                     }
 
+                    // compute and store information relating to exiting at a reward cycle
+                    let current_block_height = block_receipt.header.burn_header_height as u64;
+                    let current_reward_cycle = self
+                        .burnchain
+                        .block_height_to_reward_cycle(current_block_height)
+                        .ok_or_else(|| DBError::NotFoundError)?;
+                    let mut current_exit_at_rc = None;
+                    let mut current_proposal = None;
+                    let mut invalid_reward_cycles = vec![];
+                    let mut stacks_block_height_in_cycle = 1;
+                    let stacks_block_id =
```

Comment:
Since `stacks_block_id` is derived from `block_receipt.header.consensus_hash`, might the APIs for vote-tallying and veto-tallying make more sense if they were passed `block_hash` instead of `stacks_block_id`?  Then, they can calculate the ID hash themselves, and not have to worry about defensively checking that `consensus_hash` matches `stacks_block_id`.